pkg/storageincentives/staking/contract.go

```diff
@@ -159,37 +159,42 @@ func (s *contract) getStake(ctx context.Context, overlay swarm.Address) (*big.In
 	return abi.ConvertType(results[0], new(big.Int)).(*big.Int), nil
 }
 
-func (s *contract) DepositStake(ctx context.Context, stakedAmount *big.Int) error {
+func (s *contract) DepositStake(ctx context.Context, stakedAmount *big.Int) (txHash common.Hash, err error) {
 	prevStakedAmount, err := s.GetStake(ctx)
 	if err != nil {
-		return err
+		return
 	}
 
 	if len(prevStakedAmount.Bits()) == 0 {
 		if stakedAmount.Cmp(MinimumStakeAmount) == -1 {
-			return ErrInsufficientStakeAmount
+			err = ErrInsufficientStakeAmount
+			return
 		}
 	}
 
 	balance, err := s.getBalance(ctx)
 	if err != nil {
-		return err
+		return
 	}
 
 	if balance.Cmp(stakedAmount) < 0 {
-		return ErrInsufficientFunds
+		err = ErrInsufficientFunds
+		return
 	}
 
-	_, err = s.sendApproveTransaction(ctx, stakedAmount)
+	receipt, err := s.sendApproveTransaction(ctx, stakedAmount)
+	if receipt != nil {
```

Comment:
because there are two transactions here, I would log this one on error
and return the next transaction